geomesa-arrow/geomesa-arrow-gt/src/main/scala/org/locationtech/geomesa/arrow/io/DeltaWriter.scala

```diff
@@ -471,7 +471,7 @@ object DeltaWriter extends StrictLogging {
         val mapVector = toLoad.underlying
         val dict = dictionaries.get(sortBy)
         val sortReader = ArrowAttributeReader(sft.getDescriptor(sortBy), mapVector.getChild(sortBy), dict, encoding)
-        mergeBuilder += ((mapVector, sortReader, transfers, mappings.get(sortBy).orNull))
+        mergeBuilder += ((toLoad, sortReader, transfers, mappings.get(sortBy).orNull))
```

Comment:
Ok what's the difference between mapVector and toLoad?

Reply:
`mapVector` is the underlying arrow vector, `toLoad` is the SimpleFeatureVector that wraps it. the issue was that the SimpleFeatureVector holds a reference to an allocator, and it wasn't being passed through so it wasn't being closed.